### PR TITLE
Fix method guard in serve_data.

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

File: src/meta.jl [hunk 1]
```diff
@@ -134,10 +134,11 @@ end
 function serve_data(http::HTTP.Stream, msg, content_type)
     HTTP.setheader(http, "Content-Length" => string(length(msg)))
     HTTP.setheader(http, "Content-Type" => content_type)
+    startwrite(http)
```
Comment (staticfloat) [inline src/meta.jl:137]: Just get rid of `startwrite()` I think.

**Review by staticfloat — CHANGES_REQUESTED** [review]

File: src/meta.jl (hunk 1 above)
Comment (fredrikekre) [reply src/meta.jl:137]: Are you sure? Something like
```julia
julia> HTTP.listen("0.0.0.0", 8000) do http
           HTTP.setheader(http, "Hello" => "World")
           if http.message.method == "GET"
               HTTP.write(http, "hello")
           end
       end
```
times out `$ curl -I localhost:8000` without it.

Comment (staticfloat) [reply src/meta.jl:137]: What version of HTTP are you using?  I think this is a bug that I fixed in 0.9.1+

Comment (fredrikekre) [reply src/meta.jl:137]: I was using an older one, true. However, with HTTP#master I get 
```
┌ Error: error handling request
│   exception =
│    Server never wrote a response
│    Stacktrace:
│     [1] error(s::String)
│       @ Base ./error.jl:33
│     [2] macro expansion
│       @ ~/.julia/packages/HTTP/hbs7O/src/Servers.jl:406 [inlined]
│     [3] (::HTTP.Servers.var"#13#14"{var"#1#2", HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}, HTTP.Streams.Stream{HTTP.Messages.Request, HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})()
│       @ HTTP.Servers ./task.jl:406
└ @ HTTP.Servers ~/.julia/packages/HTTP/hbs7O/src/Servers.jl:413
```
on the server side and
```
$ curl -I localhost:8000
HTTP/1.1 500 Internal Server Error
Hello: World
```
on the client.

**Pull request merged**